rviz_rendering/src/rviz_rendering/ogre_render_window_impl.cpp

```diff
@@ -181,14 +181,14 @@ createScene(Ogre::SceneManager * ogre_scene_manager)
   */
 
   QColor color = Qt::gray;
-  new rviz_rendering::Grid(
+  std::shared_ptr<rviz_rendering::Grid>(new rviz_rendering::Grid(
```

Comment:
This is a temporary, so it will be deleted immediately which is definitely not what you want here.

Perhaps this has overlap with https://github.com/ros2/rviz/pull/165? @Martin-Idel-SI can you comment on this? It does look like a memory leak, but I don't have enough context atm to know where it should be owned and cleaned up.

Reply:
No overlap. The "real" grid is held (and properly deleted) in grid_display, which is loaded via pluginlib.

This grid belongs to the example scene (including the sphere) that was displayed with the rendering_example target. I think the whole scene gets loaded and discarded immediately in rviz, so I'm unsure whether we still need it.